### PR TITLE
pml/ucx: pml_ucx_multi_send_nb support

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**
Enable passing multi-send optimization flag for nonblocking operations.

### CODE REVIEW

**Comment by yosefe** [comment]
@brminich can you pls review?

File: ompi/mca/pml/ucx/pml_ucx_component.c [hunk 1]
```diff
@@ -79,6 +81,20 @@ static int mca_pml_ucx_component_register(void)
     ompi_pml_ucx.request_leak_check = true;
 #endif
 
+#if HAVE_DECL_UCP_OP_ATTR_FLAG_MULTI_SEND
+    ompi_pml_ucx.op_attr_nonblocking = 0;
```
Comment (brminich) [inline ompi/mca/pml/ucx/pml_ucx_component.c:85]: should we set it to 0 when no `HAVE_DECL_UCP_OP_ATTR_FLAG_MULTI_SEND` defined?

File: ompi/mca/pml/ucx/pml_ucx_component.c [hunk 2]
```diff
@@ -79,6 +81,20 @@ static int mca_pml_ucx_component_register(void)
...
+    multi_send_op_attr_enable        = 0;
+    (void) mca_base_component_var_register(&mca_pml_ucx_component.pmlm_version, "multi_send_nb",
+                                           "Enable passing multi-send optimization flag for nonblocking operations",
```
Comment (brminich) [inline ompi/mca/pml/ucx/pml_ucx_component.c:88]: but in pml it is set for blocking operations. 
what pattern does it improve?

File: ompi/mca/pml/ucx/pml_ucx.h [hunk 3]
```diff
@@ -59,6 +59,7 @@ struct mca_pml_ucx_module {
     int                       priority;
     bool                      cuda_initialized;
     bool                      request_leak_check;
+    uint32_t                  op_attr_nonblocking;
```
Comment (devreal) [inline ompi/mca/pml/ucx/pml_ucx.h:62]: Why is it `uint32_t` and not `bool`? The MCA parameter used to set it is bool (https://github.com/open-mpi/ompi/pull/9610/files#diff-6a2952e2730c99111221330017fc798631ad6603c29a257ce3bb114e588f5984R90).

**Comment by devreal** [comment]
@Artemy-Mellanox Just curious: Do you have details on what the flag `HAVE_DECL_UCP_OP_ATTR_FLAG_MULTI_SEND` does? I couldn't find it in the current UCX API document...

**Comment by Artemy-Mellanox** [comment]
> @Artemy-Mellanox Just curious: Do you have details on what the flag `HAVE_DECL_UCP_OP_ATTR_FLAG_MULTI_SEND` does? I couldn't find it in the current UCX API document...

https://github.com/openucx/ucx/blob/master/src/ucp/api/ucp.h#L715

File: ompi/mca/pml/ucx/pml_ucx.h (hunk 3 above)
Comment (Artemy-Mellanox) [reply ompi/mca/pml/ucx/pml_ucx.h:62]: In future it can hold another flags for non-blocking operations.

**Review by brminich — APPROVED** [review]

**Review by devreal — CHANGES_REQUESTED** [review]
@Artemy-Mellanox  Please squash the two commits

**Review by yosefe — APPROVED** [review]

**Review by devreal — APPROVED** [review]

**Comment by yosefe** [comment]
@Artemy-Mellanox can you pls backport to v5.0.x and v4.1.x?

**Pull request merged**